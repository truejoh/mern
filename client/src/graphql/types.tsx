import gql from 'graphql-tag';
export type Maybe<T> = T | null;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** The `Upload` scalar type represents a file upload. */
  Upload: any;
};

export type Bank = {
  __typename?: 'Bank';
  _id: Scalars['ID'];
  name: Scalars['String'];
  branchName: Scalars['String'];
};

export enum CacheControlScope {
  Public = 'PUBLIC',
  Private = 'PRIVATE',
}

export type CreateBankInput = {
  name: Scalars['String'];
  branchName: Scalars['String'];
};

export type CreateEmployeeInput = {
  name: Scalars['String'];
  number: Scalars['Int'];
  accountHolder: Scalars['String'];
  accountType?: Maybe<Scalars['String']>;
  accountNumber: Scalars['Int'];
  bank: Scalars['ID'];
};

export type Employee = {
  __typename?: 'Employee';
  _id: Scalars['ID'];
  name: Scalars['String'];
  number: Scalars['Int'];
  accountHolder: Scalars['String'];
  accountType?: Maybe<Scalars['String']>;
  accountNumber: Scalars['Int'];
  bank?: Maybe<Bank>;
};

export type EmployeesInput = {
  field?: Maybe<Scalars['String']>;
  value?: Maybe<Scalars['String']>;
  operator?: Maybe<Operator>;
};

export type Mutation = {
  __typename?: 'Mutation';
  createBank: Bank;
  updateBank: Bank;
  deleteBank: Bank;
  createEmployee: Employee;
  updateEmployee: Employee;
  deleteEmployee: Employee;
};

export type MutationCreateBankArgs = {
  bank?: Maybe<CreateBankInput>;
};

export type MutationUpdateBankArgs = {
  _id: Scalars['String'];
  bank?: Maybe<UpdateBankInput>;
};

export type MutationDeleteBankArgs = {
  _id: Scalars['String'];
};

export type MutationCreateEmployeeArgs = {
  employee?: Maybe<CreateEmployeeInput>;
};

export type MutationUpdateEmployeeArgs = {
  _id: Scalars['String'];
  employee?: Maybe<UpdateEmployeeInput>;
};

export type MutationDeleteEmployeeArgs = {
  _id: Scalars['String'];
};

export enum Operator {
  Gte = 'gte',
  Lte = 'lte',
  Eq = 'eq',
  Like = 'like',
}

export enum Order {
  Asc = 'asc',
  Desc = 'desc',
}

export type Query = {
  __typename?: 'Query';
  banks: Array<Bank>;
  employees: Array<Employee>;
};

export type QueryEmployeesArgs = {
  filters?: Maybe<Array<Maybe<EmployeesInput>>>;
  limit?: Maybe<Scalars['Int']>;
  offset?: Maybe<Scalars['Int']>;
  order?: Maybe<Order>;
  sort?: Maybe<Scalars['String']>;
};

export type UpdateBankInput = {
  name?: Maybe<Scalars['String']>;
  branchName?: Maybe<Scalars['String']>;
};

export type UpdateEmployeeInput = {
  name?: Maybe<Scalars['String']>;
  number?: Maybe<Scalars['Int']>;
  accountHolder?: Maybe<Scalars['String']>;
  accountType?: Maybe<Scalars['String']>;
  accountNumber?: Maybe<Scalars['Int']>;
  bank?: Maybe<Scalars['ID']>;
};
