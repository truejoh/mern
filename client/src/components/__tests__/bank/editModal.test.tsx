import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CreateModal from '../../bank/modals/createModal';
import { functionDeclaration } from '@babel/types';

describe('Bank Edit Modal', () => {
  it('renders component', () => {
    const props = {
      visible: true,
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<CreateModal {...props} />);

    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders props', () => {
    const props = {
      visible: true,
      onOk: () => {},
      onCancel: () => {},
    };
    const component = shallow(<CreateModal {...props} />);

    expect(component.prop('visible')).toEqual(true);
    expect(component.prop('onOk')).toBeInstanceOf(Function);
    expect(component.prop('onCancel')).toBeInstanceOf(Function);
  });
});
