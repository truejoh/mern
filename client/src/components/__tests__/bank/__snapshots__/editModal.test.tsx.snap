// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bank Edit Modal renders component 1`] = `
<Modal
  cancelText={
    <InjectIntl(InjectMassage)
      id="btn.cancel"
    />
  }
  confirmLoading={false}
  maskTransitionName="fade"
  okText={
    <InjectIntl(InjectMassage)
      id="btn.create"
    />
  }
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title={
    <InjectIntl(InjectMassage)
      id="bank.modal.title"
    />
  }
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form
    colon={true}
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
  >
    <FormItem
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="bank.modal.name"
        />
      }
    >
      <Input
        disabled={false}
        onChange={[Function]}
        placeholder="input bank name"
        type="text"
        value=""
      />
    </FormItem>
    <FormItem
      hasFeedback={false}
      label={
        <InjectIntl(InjectMassage)
          id="bank.modal.branch"
        />
      }
    >
      <Input
        disabled={false}
        onChange={[Function]}
        placeholder="input branch"
        type="text"
        value=""
      />
    </FormItem>
  </Form>
</Modal>
`;
